src/main/java/pl/allegro/tech/opel/OpelEngine.java

```diff
@@ -33,9 +35,26 @@ public OpelParsingResult parse(String expression) {
 
     public CompletableFuture<?> eval(String expression) {
         ParsingResult<OpelNode> parsingResult = getParsingResult(expression);
+        if (parsingResult.hasErrors()) {
+            throw new OpelException("Error parsing expression: '" + expression + "'" + additionalErrorMsq(parsingResult));
+        }
         return parsingResult.resultValue.getValue(embeddedEvalContext);
     }
 
+    private String additionalErrorMsq(ParsingResult<OpelNode> parsingResult) {
+        boolean hasAdditionalMeg = parsingResult.parseErrors.stream()
+                .map(ParseError::getErrorMessage)
+                .anyMatch(it -> it != null);
+        if (hasAdditionalMeg) {
```

Comment:
maybe, this is better then npe from return parsingResult.resultValue.getValue(embeddedEvalContext);